### PR TITLE
Add new ForceDelete procedures

### PR DESCRIPTION
**Pull request by yulgolem**
Add new test file new_force_delete_test.rb to check if new ForceDeleteProcedures are correct.

See #1428

### CODE REVIEW

**Comment by yulgolem** [comment]
Need to be run daily DomainCron.start_client_hold

File: app/models/domain_cron.rb [hunk 1]
```diff
@@ -78,4 +78,46 @@ def self.start_redemption_grace_period
...
+    real = 0
+
+    Domain.force_delete_scheduled.each do |domain|
+      next unless domain.client_holdable?
+
+      real += 1
+      domain.statuses << DomainStatus::CLIENT_HOLD
+      log_start_client_hold(domain)
+
+      domain.save(validate: false) and marked += 1
```
Comment (karlerikounapuu) [inline app/models/domain_cron.rb]: Maybe use `&&` ?

Comment (yulgolem) [reply app/models/domain_cron.rb]: Well, that's some semantical && ideological point here.
You see, operators 'and' and 'or' are low precedence flow control operators useful to chain methods, not boolean operators like '&&' and '||'.
See http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/ for some examples.

Comment (karlerikounapuu) [reply app/models/domain_cron.rb]: Oh, right, missed the real intention of it. Thanks for clarification

**Pull request merged**